### ISSUE TITLE
Bold, Italic and Underline formatting freezes the whole page when BoldItalicUnderlineToggles is added to toolbar

### ISSUE THREAD

**Issue by lkenny97** [mdx-editor/editor#48]
I am using Next.js 12 (pages router). I added the editor and was able to make the text bold using `CMD+B`. I then added the `BoldItalicUnderlineToggles` to the toolbar. Now when I make something bold (using the shortcut or the toolbar button) my page freezes. Here is my code:

Page file:
```
const SomeTemplate = dynamic(() => import("../app/components/templates/SomeTemplate/SomeTemplate"), {
  ssr: false
})

const Editor = dynamic(() => import("../app/components/organisms/Editor/Editor"), {
  ssr: false
})

function EditorPage() {
  const {user, loading} = useUser({redirectTo: "/"})

  if (loading || !user?.isLoggedIn)
    return <Spinner />

  return (
    <SomeTemplate>
      <Editor />
    </SomeTemplate>
  )
}

export default EditorPage
```

The Editor file:
```
import '@mdxeditor/editor/style.css'
import { MDXEditor } from '@mdxeditor/editor/MDXEditor'
import { UndoRedo } from '@mdxeditor/editor/plugins/toolbar/components/UndoRedo'
import { BoldItalicUnderlineToggles } from '@mdxeditor/editor/plugins/toolbar/components/BoldItalicUnderlineToggles'
import { toolbarPlugin } from '@mdxeditor/editor/plugins/toolbar'

interface EditorProps {}

const Editor = ({}: EditorProps) => {
  return (
    <MDXEditor 
      markdown='Hello world'
      plugins={[toolbarPlugin({
        toolbarContents: () => ( <><UndoRedo /><BoldItalicUnderlineToggles /></>)
      })]}
    />
  );
};

export default Editor;
```

Is there something I am missing? I don't have any controlled logic to lead to an infinite loop. Why is adding `BoldItalicUnderlineToggles` breaks the working functionality?

**Comment by petyosi** [comment]
Is this by any chance https://github.com/mdx-editor/editor/issues/46?

**Comment by lkenny97** [comment]
@petyosi Not sure. The effect is the same at least :D  
Is it really related to legacy mode though?

There are some weird points that I want to point out in case if it helps:

For JSX, this works fine when there is only one element inside the toolbar
```
return (
    <MDXEditor
      markdown='Hello world'
      plugins={[
        jsxPlugin({ jsxComponentDescriptors }),
        toolbarPlugin({
          toolbarContents: () => ( <><InsertMyDiv /></>)
        })]}
    />
  );
```

But as soon as I add something else like
```
...
toolbarContents: () => ( <><UndoRedo /><InsertMyDiv /></>)
...
```
The page freezes when I type something inside of the field provided
![Screenshot 2023-08-28 at 17 36 54](https://github.com/mdx-editor/editor/assets/32220582/fd8c27ff-007f-44d7-b52b-a09a781022bd)
Doesn't freeze when I type below it :D 

For `BoldItalicUnderlineToggles` however, it freezes even if it is the only thing in the toolbar.

I am so confused, been trying to debug this the whole day

**Comment by petyosi** [comment]
@lkenny97 you can check if your app is using the legacy rendering mode in your main file. I found one potential problem and solved in 0.15.2, please test and let me know.
